server/controllers/admin.py

```diff
@@ -637,9 +637,19 @@ def enrollment(cid):
     return render_template('staff/course/enrollment.html',
                            enrollments=students, staff=staff,
                            single_form=single_form,
+                           unenroll_form=forms.CSRFForm(),
                            courses=courses,
                            current_course=current_course)
 
+@admin.route("/course/<int:cid>/<string:user_id>/unenroll", methods=['POST'])
+@is_staff(course_arg='cid')
+def unenrollment(cid, user_id):
+    user = User.query.filter_by(id=user_id).one_or_none()
+    if user and user.is_enrolled(cid):
```

Comment:
I managed to get a 500 by opening the enrollment page in two different tabs and trying to delete the same user from both of them